tutorials/tts/FastPitch_MultiSpeaker_Pretraining.ipynb

```diff
@@ -0,0 +1,867 @@
+    "os.makedirs(suppdir, exist_ok=True)\n",
+    "os.makedirs(pitch_dir, exist_ok=True)\n",
+    "\n",
+    "train_pitchs = []\n",
+    "train_datas = json_reader(train_manifest)\n",
+    "base_data_dir = get_base_dir([item[\"audio_filepath\"] for item in train_datas])\n",
+    "for m in tqdm(train_datas): train_pitchs.append(get_pitch(m))\n",
+    "    \n",
+    "valid_datas = json_reader(valid_manifest)\n",
+    "base_data_dir = get_base_dir([item[\"audio_filepath\"] for item in valid_datas])\n",
+    "for m in tqdm(valid_datas): get_pitch(m)\n",
+    "\n",
+    "train_pitchs = np.concatenate(train_pitchs)\n",
+    "pitch_mean = float(np.mean(train_pitchs))\n",
+    "pitch_std = float(np.std(train_pitchs))\n",
+    "\n",
+    "with open(os.path.join(manidir, 'pitch_stats.json'), 'w') as f:\n",
+    "    json.dump({'pitch':[pitch_mean, pitch_std]}, f)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "a41ee532",
+   "metadata": {},
+   "source": [
+    "## c. Training"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "7f954beb",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!mkdir -p {confdir} && cd {confdir} && wget https://raw.githubusercontent.com/NVIDIA/NeMo/$BRANCH/examples/tts/conf/fastpitch_align_44100_adapter.yaml"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "e82e6860",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!mkdir -p {codedir} && cd {codedir} && wget https://raw.githubusercontent.com/NVIDIA/NeMo/$BRANCH/examples/tts/fastpitch.py"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "12307264",
+   "metadata": {},
+   "source": [
+    "### Important notes\n",
+    "* [Data] **speaker_id** in **sup_data_types**: each data has an unique speaker index (start from 0) in the input.\n",
+    "* [Data] **reference_audio** in **sup_data_types**: each data has a reference audio (from the same speaker) in the input.\n",
+    "* [Speaker] **model.speaker_encoder.lookup_module**: model creates lookup table to get speaker embedding from speaker id.\n",
+    "* [Speaker] **model.speaker_encoder.lookup_module.n_speakers**: model gets the speaker size. \n",
+    "* [Speaker] **model.speaker_encoder.gst_module**: model creates global style token to extract speaker information from reference audio.\n",
+    "* [Condition] **condition_types=\"['add', 'layernorm']\"**: insert conditions with `add` operation to inputs and `layernorm` operation to layernorms."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "6ffdce37",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Normally 200 epochs\n",
+    "!(python {codedir}/fastpitch.py \\\n",
+    "--config-path={os.path.abspath(confdir)} \\\n",
+    "--config-name=fastpitch_align_44100_adapter.yaml \\\n",
+    "+init_from_pretrained_model=\"tts_en_fastpitch\" \\\n",
+    "sample_rate={sample_rate} \\\n",
```

Comment:
Should avoid providing sample_rate as an override since the config `fastpitch_align_44100_adapter.yaml` is for 44khz specifically.